src/generic/Matrix.jl

```diff
@@ -397,6 +397,38 @@ function Base.view(M::AbstractAlgebra.MatElem{T}, rows::Colon, cols::Colon) wher
    return view(M, 1:nrows(M), 1:ncols(M))
 end
 
+###############################################################################
+#
+#   Block replacement
+#
+###############################################################################
+
+function setindex!(a::AbstractAlgebra.MatElem{T}, b::AbstractAlgebra.MatElem{T}, r::UnitRange{Int}, c::UnitRange{Int}) where T
+    _checkbounds(a, r, c)
+    size(b) == (length(r), length(c)) || throw(DimensionMismatch("tried to assign a $(nrows(b))x$(ncols(b)) matrix to a $(length(r))x$(length(c)) destination"))
+    startr = first(r)
+    startc = first(c)
+    for i in r
+        for j in c
+            a[i, j] = deepcopy(b[i - startr + 1, j - startc + 1])
```

Comment:
I'm not sure if the deepcopy is actually required here. I think we allow matrices to alias each other's elements (mostly).

Reply:
I followed the example set by `sub`, where the entries are also deepcopied.